bothan-core/src/telemetry/server.rs

```diff
@@ -0,0 +1,49 @@
+use std::error::Error;
+use std::net::{SocketAddr, ToSocketAddrs};
+use std::sync::Arc;
+
+use axum::{Extension, Router};
+use axum::response::IntoResponse;
+use axum::routing::get;
+
+use prometheus::{Encoder, Registry, TextEncoder};
+use tokio::task::JoinHandle;
+
+pub type BoxError = Box<dyn Error + Send + Sync>;
+
+pub fn spawn_server<A>(
+    addr: A,
+    registry: Arc<Registry>,
+) -> Result<(SocketAddr, JoinHandle<Result<(), BoxError>>), BoxError>
```

Comment:
Please do not use box error unless necessary, make a custom error or in the worse case use anyhow